shared/applicability/package.yml

```diff
@@ -0,0 +1,9 @@
+name: "cpe:/a:{arg}"
+title: "Package {arg} is installed"
+check_id: cond_package_{arg}
+template:
+  name: cond_package
```

Comment:
Because it is a "conditional" test. We use this term everywhere.

Reply:
What does a "conditional" test mean?

Reply:
It means the same thing as in CPE entry fields `bash_conditional` and `ansible_conditional`: https://github.com/ComplianceAsCode/content/blob/06edcfef2eff4e6a8bb0cac6b27ec98f1d5479f1/shared/applicability/machine.yml#L4

Reply:
I agree with Jan, and BTW the `machine.yml` also shows this: `bash_conditional` is a variable that stores a Bash conditional statement, so the naming is accurate - you can store many things in Bash variables, not only conditional statements. Whereas an OVAL check ID is always some sort of test or condition, so in the `machine yml`, it is stored under `check_id`, and the name `installed_env_is_a_machine` describes what the check is about. Therefore, why not to pick a check ID `{arg}_package_installed` or something similar?
Also I would factor in the `installed` either in the template name or in template arguments, as it is likely that we will get platform checks checking on absence of packages in the future.

Reply:
The absence of package is just `not package[gdm]` which will end up as negated condition on respective levels.

The `check_id` is a just a reference to OVAL check, which now could be a shared static check or generated from a template. They are used for building CPE AL logical expression for the platform.

The set of `bash_conditional` and `ansible_conditional` are on the other hand entities that contain the whole conditional expression for individual platform element (symbol), that would be glued together with boolean operators (for the whole platform expression) and added directly to every remediation affected by this platform expression on the stage of remediation rendering.

Reply:
I'm okay to rename template into whatever: `platform_`, `inventory_` (because it is the only difference on the OVAL side), `cpe_`. The only things that worries me here is the logical connection with `*_conditional` names.

Reply:
The final form for applicability entity in `cpe_versioning` implementation is:
```yaml

name: 'cpe:/a:{arg}:{ver_cpe}'
title: 'Package {arg} {ver_str} is installed'
template:
    name: cond_package
args:
    chronytest:
        title: "Chrony Server Test"
        pkgname: chrony
        pkgname@debian10: chrony
    ntptest:
        title: "NTP Server Test"
        pkgname: ntp
```
or
```yaml
name: cpe:/a:container
title: Container
conditional:
    bash: '[ ! -f /.dockerenv ] && [ ! -f /run/.containerenv ]'
    oval_id: installed_env_is_a_container
```
In the first case everything is generated by the template (including the check_id, because template knows how to id itself). And in the second case everything is defined without the template (using static OVAL check file, hence the renamed oval_id).

The combination of templated and direct definitions was also possible, with direct definitions taking over template content on individual lang basis.

Reply:
This requires mass renaming of like 100 files and I'm not sure if it even makes any sense.

Reply:
> This requires mass edit of like 100 files and I'm not sure if it even makes any sense.

I don't want to rename 100 files. Instead, I try to point out that the comment above is misleading because the actual contents of the file(s) is different.

Reply:
Okay, `platform_package` it is.